### PR TITLE
Add clang-format and workflow for consistency #21

### PR DESCRIPTION
**Pull request by nikhiljangra264**
#21 
1. Added clang-format rules to ensure consistency across source files.
1. Added a workflow test clang-format.yml to check code formatting on pull requests and when changes are made to the master branch.

### CODE REVIEW

**Comment by fffelix-huang** [comment]
Thanks! I'll be reviewing within couple days.

**Review by fffelix-huang — CHANGES_REQUESTED** [review]
Hi

There are some small changes to be made.

File: .clang-format [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+AllowShortEnumsOnASingleLine: false
+AllowShortIfStatementsOnASingleLine: false
+AlwaysBreakTemplateDeclarations: Yes
+BasedOnStyle: WebKit
+BitFieldColonSpacing: After
+BinPackParameters: false
+BreakBeforeBinaryOperators: NonAssignment
+BreakBeforeBraces: Custom
+BraceWrapping:
+  AfterFunction: false
```
[the remark this answers is not in the document]
Comment (nikhiljangra264) [reply .clang-format:19]: which one do you prefer?
1. value:false
```
namespace name {
class class_name {
   public:
    function_name() {
        if () {
        } else { 
        }
    }
}
}
```
2. value:true
```
namespace name 
{
class class_name 
{
   public:
    function_name() 
    {
        if () 
        {
        }
        else
        {
        }
    }
}
}
```

or combination

**Comment by nikhiljangra264** [comment]
Hello @fffelix-huang, I have made some changes requested

**Comment by nikhiljangra264** [comment]
apart from that I have added clang-format off and on at places like this
```cpp
// clang-format off
constexpr std::pair<Value, Value> PieceSquareTable[PieceType::PIECE_NB][Square::SQ_NB] = {
    { // Pawn
        {0, 0}, {0, 0}, {0, 0}, {0, 0}, {0, 0}, {0, 0}, {0, 0}, {0, 0},
        {166, 256}, {192, 256}, {204, 256}, {216, 256}, {216, 256}, {204, 256}, {192, 256}, {166, 256},
        {166, 256}, {192, 256}, {210, 256}, {242, 256}, {242, 256}, {210, 256}, {192, 256}, {166, 256},
        {166, 256}, {192, 256}, {220, 256}, {268, 256}, {268, 256}, {220, 256}, {192, 256}, {166, 256},
        {166, 256}, {192, 256}, {220, 256}, {242, 256}, {242, 256}, {220, 256}, {192, 256}, {166, 256},
        {166, 256}, {192, 256}, {210, 256}, {216, 256}, {216, 256}, {210, 256}, {192, 256}, {166, 256},
        {166, 256}, {192, 256}, {204, 256}, {216, 256}, {216, 256}, {204, 256}, {192, 256}, {166, 256},
        {0, 0}, {0, 0}, {0, 0}, {0, 0}, {0, 0}, {0, 0}, {0, 0}, {0, 0},
    },
}
```
to avoid formating on this code even if it exceeds the column width 100

**Comment by fffelix-huang** [comment]
Great! This is better than I expected :)
One last request is to add formatting to Makefile, (`make format` should apply clang-format all source codes including cpps and hpps).
After this I'll merge this PR!

**Comment by nikhiljangra264** [comment]
I have Added it. so we can run the `make format` command to format code.
Please take a look

**Pull request merged**